be/src/storage/lake/tablet_writer.h

```diff
@@ -33,12 +33,26 @@ enum WriterType : int { kHorizontal = 0, kVertical = 1 };
 // Basic interface for tablet writers.
 class TabletWriter {
 public:
-    explicit TabletWriter(Tablet tablet, std::shared_ptr<const TabletSchema> tschema)
-            : _tablet(tablet), _schema(std::move(tschema)) {}
+    explicit TabletWriter(Tablet tablet, std::shared_ptr<const TabletSchema> schema, int64_t txn_id)
+            : _tablet(tablet), _schema(std::move(schema)), _txn_id(txn_id) {}
 
     virtual ~TabletWriter() = default;
 
-    virtual int64_t tablet_id() const { return _tablet.id(); }
+    int64_t tablet_id() const { return _tablet.id(); }
+
+    int64_t txn_id() const { return _txn_id; }
+
+    // Return a list of files generated by this writer.
+    // The file paths are relative to the tablet group path.
+    //
+    // PREREQUISITES: the writer has successfully `finish()`ed but not yet `close()`ed.
+    std::vector<std::string> files() const { return _files; }
```

Comment:
Why not returning a const reference?

Reply:
In theory, it's copied. In reality, no, most modern compilers take advantage of [return value optimization](http://en.wikipedia.org/wiki/Return_value_optimization).

Reply:
This is the original implementation, change the signature requires many code changes, because many places use the return value as below:
```
for (auto& f : writer->files()) {
    op_log->add_segment(std::move(f));
}
```
And based on the usages like above, return const reference will disable the `std::move` on the elements of the vector.

Reply:
If you do think return const reference is necessary, I can open another PR to change it.